### PR TITLE
replace ESCount with CountNode/CountOperation

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/count/InternalCountOperation.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        this.indicesService = indicesService;
+    }
+
+    @Override
+    public long count(Map<String, ? extends Collection<Integer>> indexShardMap, WhereClause whereClause) throws IOException {
+        long count = 0L;
+        for (Map.Entry<String, ? extends Collection<Integer>> entry : indexShardMap.entrySet()) {
+            String index = entry.getKey();
+            for (Integer shardId : entry.getValue()) {
+                count += count(index, shardId, whereClause);
```
Comment (mfelsche) [inline sql/src/main/java/io/crate/operation/count/InternalCountOperation.java]: are you sure this should be done sequentially?

File: sql/src/main/java/io/crate/planner/consumer/CountConsumer.java [hunk 2]
```diff
@@ -56,29 +67,46 @@ public boolean consume(AnalyzedRelation rootRelation, ConsumerContext context) {
...
             }
 
-            if (firstNonNull(table.querySpec().limit(), 1) < 1 ||
-                    table.querySpec().offset() > 0){
+            if (firstNonNull(querySpec.limit(), 1) < 1 ||
+                    querySpec.offset() > 0){
                 return new NoopPlannedAnalyzedRelation(table);
             }
 
+            // TODO: filterRouting
```
Comment (mfelsche) [inline sql/src/main/java/io/crate/planner/consumer/CountConsumer.java]: TODO?

File: sql/src/test/java/io/crate/operation/count/InternalCountOperationTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import io.crate.operation.operator.EqOperator;
+import io.crate.planner.symbol.Literal;
+import io.crate.test.integration.CrateIntegrationTest;
+import io.crate.testing.TestingHelpers;
+import io.crate.types.DataTypes;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+
+@CrateIntegrationTest.ClusterScope(scope = CrateIntegrationTest.Scope.SUITE, numNodes = 1)
```
Comment (mfelsche) [inline sql/src/test/java/io/crate/operation/count/InternalCountOperationTest.java]: what about using CrateSingleNodeTest?

Comment (mfussenegger) [inline sql/src/test/java/io/crate/operation/count/InternalCountOperationTest.java]: that one doesn't have execute() and our other cluster setup stuff

**Comment by mfelsche** [comment]
lgtm

**Pull request merged**